middle_end/flambda/simplify/simplify_expr.rec.ml

```diff
@@ -1196,11 +1277,15 @@ and simplify_function_call
         | Some newer -> Rec_info.merge rec_info ~newer
       in
       let callee's_code_id_from_type = I.code_id inlinable in
+      let must_be_detupled =
+        I.is_tupled inlinable &&
+        calling_convention_from_call_kind = `Closure
```

Comment:
Maybe add a comment explaining why de-tupling must only occur in this single case.

Reply:
Done, together with a small refactoring to avoid using a polymorphic variant.